service/src/test/java/uk/nhs/adaptors/gp2gp/DeterministicIdGeneratorService.java

```diff
@@ -0,0 +1,30 @@
+package uk.nhs.adaptors.gp2gp;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.LinkedList;
+import java.util.Queue;
+
+import lombok.SneakyThrows;
+import uk.nhs.adaptors.gp2gp.common.service.RandomIdGeneratorService;
+
+public class DeterministicIdGeneratorService extends RandomIdGeneratorService {
+
+    Queue<String> ids;
+
+    @SneakyThrows
+    public void reset() {
+        ids = new LinkedList<>();
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(this.getClass().getResourceAsStream("/uuids.txt")))) {
+            reader.lines().forEach(ids::add);
+        }
```

Comment:
Might be clearer?
```suggestion
        Files.lines(Path.of("src", "test", "resources", "uuids.txt")).forEach(ids::add);
```

Reply:
It is not possible to access classpath resources this way